### DISCUSSION TITLE
How to turn using gyro

### DISCUSSION THREAD

**Discussion by aiconnell** [pybricks/support#1442] [Q&A]
![IMG_1635](https://github.com/pybricks/support/assets/159912441/8c92033c-94b5-4a11-88d7-31e2978e55c2)
I coach a small robotics team. We’ve been using Spike prime word blocks. However, they wanted to try some of the powered up motors, and that does not work with the spike prime software. I installed Pybricks last night and started playing around to see how much of a change it would be. I could figure out the basics, but I could not figure out how to turn using Gyro using blocks. Can anyone help? I’m trying to see if there’s command similar to what they understand from spike, attached photo for example.

**Comment by laurensvalk** [comment]
While you could do exactly as you did above, there is also a much simpler way:

![image](https://github.com/pybricks/support/assets/12326241/93e5a1ec-aa4b-4af1-9e8a-d29888598739)

This way gyro correction is built into the movements. Try pushing the robot around -- it should self correct. When this is unwanted (such a when re-aligning against a wall), set the light blue block above back to using the rotation sensors.

If you prefer the manual approach, you could read the heading angle and build your own movements:

![image](https://github.com/pybricks/support/assets/12326241/ef9f7427-3e7f-40ef-af76-5b7f6814aa38)

**Comment by aiconnell** [comment]
I tried it, but it doesn’t turn the correct amount. Not sure why not? Always short by a lot. So to get it to turn 180 degrees had to set it to 235 degrees

**Reply by laurensvalk** [reply]
Can you share your full program? Do the dimensions in the drive base blocks match your robot?
